meson.build

```diff
@@ -0,0 +1,150 @@
+project('GMime',
+	['c'],
+	license: 'GPLv2',
+    version: '3.2.13',
+    meson_version: '>=1.0.0',
+    default_options : [
+    ])
+
+gmime_version = meson.project_version()
+ver_arr = gmime_version.split('.')
+
+GMIME_MAJOR_VERSION=ver_arr[0].to_int()
+GMIME_MINOR_VERSION=ver_arr[1].to_int()
+GMIME_MICRO_VERSION=ver_arr[2].to_int()
+GMIME_BINARY_AGE= 100 * GMIME_MINOR_VERSION + GMIME_MICRO_VERSION
+GMIME_INTERFACE_AGE=0
+
+gmime_package = 'gmime-@0@.0'.format(GMIME_MAJOR_VERSION)
+
+add_global_arguments('-DHAVE_CONFIG_H=1', language : 'c')
+
+iconv = dependency('iconv', required : false, method: get_option('iconv'))
+libidn2 = dependency('libidn2', version: '>=2.0.0', required : false)
+gpgme = dependency('gpgme', version: '>=1.6.0', required: get_option('crypto'))
+glib =  dependency('glib-2.0', version: '>=2.68')
+gobject = dependency('gobject-2.0')
+deps = [
+  dependency('zlib', version: '>=1.2.5.2'),
+  glib,
+  gpgme,
+  gobject,
+  dependency('gio-2.0'),
+  libidn2,
+  iconv,
+]
+
+gmime_prefix = get_option('prefix')
+gmime_datadir = join_paths(gmime_prefix, get_option('datadir'))
+
+cc = meson.get_compiler('c')
+
+conf_data = configuration_data()
+conf_data.set_quoted('GMIME_VERSION', meson.project_version())
+
+LIBS = ['glib-2.0', 'gio-2.0', 'gobject-2.0' ]
+EXTRA_LIBS = ['zlib']
+
+if cc.has_header('getopt.h')
+  if cc.has_function('getopt_long')
+    conf_data.set('HAVE_GETOPT_H', 1)
+  endif
+endif
+
+if gpgme.found()
+  conf_data.set('ENABLE_CRYPTO', 1)
+  EXTRA_LIBS += 'gpgme'
+  EXTRA_LIBS += 'gpg-error'
+endif
+
+if libidn2.found()
+  conf_data.set('LIBIDN', 1)
+  EXTRA_LIBS += 'libidn2'
+endif
+
+
+iconv_detect = '''
+#define CONFIGURE_IN
+#include "iconv-detect.c"
+'''
+
+if get_option('have_iconv_detect_h')
+  conf_data.set('HAVE_ICONV_DETECT_H', 1)
+else
+  # not cross-compiler secure
+  root = include_directories('.')
+  has_detect = cc.run(iconv_detect, name: 'iconv-detect', 
+    include_directories: root).returncode() == 0
+  if has_detect
+    conf_data.set('HAVE_ICONV_DETECT_H', 1)
+  endif
+endif
+
+foreach hd : [
+    { 'm': 'HAVE_DLFCN_H',            'v': 'dlfcn.h', },
+    { 'm': 'HAVE_INTTYPES_H',         'v': 'inttypes.h', },
+    { 'm': 'HAVE_NETDB_H',            'v': 'netdb.h', },
+    { 'm': 'HAVE_STDINT_H',           'v': 'stdint.h', },
+    { 'm': 'HAVE_WINDOWS_H',          'v': 'windows.h'},
+    { 'm': 'HAVE_SYS_MMAN_H',         'v': 'sys/mman.h', },
+    { 'm': 'HAVE_SYS_PARAM_H',        'v': 'sys/param.h', },
+  ]
+  if cc.has_header(hd['v'])
+    conf_data.set10(hd['m'], true)
+  endif
+endforeach
```

Comment:
This could be a flat list of headers. e.g. if you use Meson's `tools/ac_converter.h`, it suggests the following:

```meson
cc = meson.get_compiler('c')
cdata = configuration_data()
check_headers = [
  'dlfcn.h',
  'getopt.h',
  'iconv/detect.h',
  'inttypes.h',
  'netdb.h',
  'poll.h',
  'stdint.h',
  'stdio.h',
  'stdlib.h',
  'strings.h',
  'string.h',
  'sys/mman.h',
  'sys/param.h',
  'sys/stat.h',
  'sys/types.h',
  'time.h',
  'unistd.h',
  'winsock2.h',
]

foreach h : check_headers
  if cc.has_header(h)
    cdata.set('HAVE_' + h.underscorify().to_upper(), 1)
  endif
endforeach

check_functions = [
  ['HAVE_FSYNC', 'fsync', '#include<unistd.h>'],
  ['HAVE_GETADDRINFO', 'getaddrinfo', '#include<netdb.h>'],
# check token ['HAVE_GETDOMAINNAME']
# check token ['HAVE_GETHOSTNAME']
  ['HAVE_GETPAGESIZE', 'getpagesize', '#include<unistd.h>'],
  ['HAVE_MMAP', 'mmap', '#include<sys/mman.h>'],
# check token ['HAVE_MSYNC']
# check token ['HAVE_MUNMAP']
  ['HAVE_POLL', 'poll', '#include<poll.h>'],
  ['HAVE_SELECT', 'select', '#include<sys/select.h>'],
# check token ['HAVE_UTSNAME_DOMAINNAME']
]

foreach f : check_functions
  if cc.has_function(f.get(1), prefix : f.get(2))
    cdata.set(f.get(0), 1)
  endif
endforeach

cdata.set('SIZEOF_OFF_T', cc.sizeof('off t'))
cdata.set('SIZEOF_SIZE_T', cc.sizeof('size_t'))
cdata.set('SIZEOF_SSIZE_T', cc.sizeof('ssize_t'))
cdata.set('SIZEOF_TIME_T', cc.sizeof('time t'))

configure_file(input : 'config.h.meson',
  output : 'config.h',
  configuration : cdata)
```

Note the use of:
```meson
cdata.set('HAVE_' + h.underscorify().to_upper(), 1)
```

Reply:
Cool, never heard of that tool. Fixed